### PR TITLE
Add specification

### PR DESCRIPTION
**Pull request by ewilligers**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            information about streams.
+          </div>
+        </section>
+        <section data-dfn-for="UDPSocket">
+          <h3>
+            `UDPSocket` interface
+          </h3>
+          <pre class="idl">
+          [Exposed=Window, SecureContext] interface UDPSocket {
+            async iterable&lt;SocketOptions, Blob&gt;;
```
Comment (ewilligers) [inline index.html:143]: Only remoteAddress and remotePort are relevant from SocketOptions. We could define a new dictionary.

They will be important if/when we add more general UDP support. Unix [recvfrom() / recvmsg()](https://linux.die.net/man/2/recvfrom) and Chromium [UDPSocketListener](https://source.chromium.org/chromium/chromium/src/+/master:services/network/public/mojom/udp_socket.mojom;drc=71251a724053fd101e07f664914f06b9b4372038;l=225) provide this information with each incoming datagram.

**Comment by ewilligers** [comment]
[Preview](https://ewilligers.github.io/raw-sockets/index.html)

File: index.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            boolean noDelay;
+          };
+        </pre>
+          <p>
+            The <dfn>SocketOptions</dfn> dictionary consists of several
+            optional members:
+          </p>
+          <dl data-sort="">
+            <dt>
+              <dfn>localAddress</dfn> member
```
Comment (phoglenix) [inline index.html:82]: Add a little description for what each option does/means?

File: index.html [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            Promise&lt;undefined&gt; close();
+
+            readonly attribute DOMString remoteAddress;
+            readonly attribute unsigned short remotePort;
+            readonly attribute ReadableStream readable;
+            readonly attribute WritableStream writable;
+          };
+        </pre>
+          <div class="note">
+            The only current implementation (Chromium) does not support "byob"
```
Comment (phoglenix) [inline index.html:132]: Link to BYOB definition? https://streams.spec.whatwg.org/#enumdef-readablestreamreadermode

File: index.html [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        // eventually add mdn: true, and caniuse: "direct-sockets"
+      };
+    </script>
+  </head>
+  <body data-cite="Direct Sockets">
+    <section id="abstract">
+      <p>
+        This specification defines an API that allows web applications to talk
+        to servers and devices that have their own protocols incompatible with
+        what’s available on the web.
```
Comment (phoglenix) [inline index.html:35]: super-nit: tone of "what's" is informal. Perhaps replace with "protocols" or "those".

File: index.html (hunk 1 above)
Comment (phoglenix) [reply index.html:143]: Seems fine to use SocketOptions then. Maybe note here if particular fields are required/expected to be populated?

Comment (ewilligers) [reply index.html:182]: done

File: index.html (hunk 2 above)
Comment (ewilligers) [reply index.html:94]: done

File: index.html (hunk 3 above)
Comment (ewilligers) [reply index.html]: done

File: index.html (hunk 4 above)
Comment (ewilligers) [reply index.html]: those

**Review by phoglenix — APPROVED** [review]
LGTM

**Pull request merged**